### ISSUE TITLE
[BUG] Critical Glib assertion failed using GTK-64

### ISSUE THREAD

**Issue by ramongonze** [arx-deidentifier/arx#302]
**Describe the bug**
When opening the ARX GUI on Linux, it keeps (in loop) showing the following messages in the terminal:

```
(ARX Anonymization Tool:15507): GLib-GObject-WARNING **: 15:56:47.155: ../../../gobject/gclosure.c:729: unable to remove uninstalled invalidation notifier: 0x7ffa58621ec0 (0x7ffa8d028890)
(ARX Anonymization Tool:15507): GLib-GObject-CRITICAL **: 15:57:25.649: g_closure_add_invalidate_notifier: assertion 'closure->n_inotifiers < CLOSURE_MAX_N_INOTIFIERS' failed
```
Obs: 15507 = PID number

Besides in random moments the program crashes (the interface freezes and becomes unusable) and returns the following error:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007ffa58634610, pid=15507, tid=0x00007ffa90f32700
#
# JRE version: OpenJDK Runtime Environment (8.0_242-b08) (build 1.8.0_242-8u242-b08-0ubuntu3~19.10-b08)
# Java VM: OpenJDK 64-Bit Server VM (25.242-b08 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libgobject-2.0.so.0+0x39610]  g_type_check_instance_is_fundamentally_a+0x20
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/ramon/Desktop/arx/jars/hs_err_pid15507.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```
[hs_err_pid15507.log](https://github.com/arx-deidentifier/arx/files/4308360/hs_err_pid15507.log)

**To Reproduce**
1. Clone the repository
2. Build it using ant
3. Go to folder **jars** and execute "java -jar arx-3.8.0-gtk-64.jar"

**Expected behavior**
It was expected no erros/warnings and no crashes.

**ARX GUI (please complete the following information):**
OS' tested:
- Linux Ubuntu 19.10 5.3.0-40-generic
- Arch Linux 5.5.8-arch1-1

**Additional context**
- I have searched about the problem and found that the problem could be related to gnome or the theme, but I have changed the theme to all possible options and it didn't work.
- I have tried to set **ulimit -c unlimited**, but it didn't change anything.
- I have tried to build using jdk-8 and 11, but both keep the problem.

**Comment by nartz** [comment]
What themes on each of these OS'es have you tried?

**Comment by ramongonze** [comment]
> What themes on each of these OS'es have you tried?

Ubuntu: Adwaita (default), Adwaita-dark, Yaru, Yaru-dark, Yaru-light
Arch: Arch

**Comment by ramongonze** [comment]
> Is this the bug you identified? This seems to be related:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=419729
If this is truly the bug, then currently it looks like ant is using swt 4.2.1, I believe this looks like it was fixed in 4.3.X.

It presents the same critical error as the one I reported. I tried to replace swt-4.2.1 by swt-4.3.2 and built arx, but the problem kept. 

> Workarounds mentioned are to try to export some environment variables (make sure to set them properly so they 'take' and are probably set globally so the process can read them as well). Have you tried setting these different environment variables mentioned as workarounds?
export SWT_GTK3=0 or SWT_GTK3=1
UBUNTU_MENUPROXY=0
env LIBOVERLAY_SCROLLBAR=0 SWT_GTK3=0 /thepathToEclipse/eclipse

I have tried all of these workarounds, but none of them worked.

> Another workaround might be to try to edit the current themefile, for the widgets responsible:
https://askubuntu.com/questions/513471/kubuntu-14-04eclipse-adt-crashes-at-button-ok-from-project-properties/530468#530468

As the assertion error I get does not respect to specific widgets, I could not edit themefiles.

---

Besides the error is not directly related to gnome, because I have also tested in Linux Mint (with cinnamon).

---

The arx-3.7.0 from branch origin/swt-4-7-2 eliminates both erros presented in this issue here, and a new one appeared tough:

`(SWT:24876): Gtk-WARNING **: 10:51:10.313: Negative content width -2 (allocation 11, extents 6x7) while allocating gadget (node separator, owner GtkSeparatorToolItem)`
 
However, it seems to be related to a specific widget (GtkSeparatorToolItem), and it may be easier to fix. 

So it seems that swt-4.7.2 solves the problem.
Do you have a forecast for merging branch swt-4-7-2 with master?

**Comment by prasser** [comment]
We are now actively working on this, as we need to update all GUI libraries anyway, because the current version of ARX doesnt work on MacOS Catalina. A fix will be released with the next version.

**Comment by prasser** [comment]
SWT updates have been merged into master. Closing this issue now. Please open further issues, should any problems remain. Thanks again for your input!
